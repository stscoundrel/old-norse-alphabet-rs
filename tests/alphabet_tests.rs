use old_norse_alphabet::{get_lowercase, get_uppercase};
use old_norse_alphabet::{ETH, THORN, O_CAUDATA, SLASHED_O, AE, OE, ASH};
use insta::assert_json_snapshot;

#[test]
fn exposes_special_characters() {
    let mut special_characters = "".to_string();

    special_characters.push_str(&ETH.to_string());
    special_characters.push_str(&THORN.to_string());
    special_characters.push_str(&O_CAUDATA.to_string());
    special_characters.push_str(&ASH.to_string());
    special_characters.push_str(&SLASHED_O.to_string());
    special_characters.push_str(&OE.to_string());
    special_characters.push_str(&AE.to_string());

    let expected = "ðþǫæøœæ".to_string();

    assert_eq!(special_characters, expected);
}

#[test]
fn exposes_lower_alphabet() {
    let lower = get_lowercase().to_vec();

    assert_json_snapshot!(lower);
}

#[test]
fn exposes_upper_alphabet() {
    let upper = get_uppercase().to_vec();

    assert_json_snapshot!(upper);
}