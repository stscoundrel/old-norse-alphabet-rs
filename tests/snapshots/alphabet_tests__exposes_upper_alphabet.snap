---
source: tests/alphabet_tests.rs
expression: upper

---
[
  "A",
  "Á",
  "B",
  "D",
  "Ð",
  "E",
  "É",
  "F",
  "G",
  "H",
  "I",
  "Í",
  "J",
  "K",
  "L",
  "M",
  "N",
  "O",
  "Ó",
  "P",
  "R",
  "S",
  "T",
  "U",
  "Ú",
  "V",
  "W",
  "Y",
  "Ý",
  "Þ",
  "Æ",
  "Ǫ",
  "Ø",
  "Œ"
]
